pkg/nimble/rpble/include/nimble_rpble.h

```diff
@@ -0,0 +1,178 @@
+ * number of free BLE connection slots. In case where multiple nodes advertise
+ * the same RPL rank, the one with the largest number of open BLE connection
+ * slots is selected. The idea behind this is to balance the number of BLE
+ * connections per node, and with that also to balance the RPL DODAG.
+ *
+ *
+ * # Usage
+ * To use this module, simply include `nimble_rpble` into your build. If nothing
+ * is explicitly configured, the module will use the default configuration as
+ * specified in `pkg/nimble/rpble/include/nimble_rpble_params.h`.
+ *
+ * Once a node is configured as RPL root (e.g. using the `rpl root ..` shell
+ * command), it will automatically start to advertise itself. Non-RPL-root nodes
+ * will automatically scan for potential parent nodes and join the network as
+ * soon as they find fitting neighbors.
+ *
+ *
+ * ## Implementation status
+ * In its current state, the implementation only works for environments where a
+ * single RPL network with a single DODAG are present. The DODAG ID, instance
+ * ID, and DODAG version are currently pretty much ignored when scanning for
+ * potential parents.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface for the nimble_rpble module
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef NIMBLE_RPBLE_H
+#define NIMBLE_RPBLE_H
+
+#include "nimble_netif.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   rpble configuration parameters
+ */
+typedef struct {
+    uint32_t scan_itvl;         /**< scan interval when scanning for parents */
+    uint32_t scan_win;          /**< scan window when scanning for parents */
+    uint32_t adv_itvl;          /**< advertising interval used when advertising
+                                 *   RPL context to child nodes */
+    uint32_t conn_scan_itvl;    /**< scan interval when connecting to parent */
+    uint32_t conn_scan_win;     /**< scan window when connecting to parent */
+    uint32_t conn_scan_to;      /**< timeout when connecting to parent */
+    uint32_t conn_itvl_min;     /**< lower bound of connection interval range */
+    uint32_t conn_itvl_max;     /**< upper bound of connection interval range */
+    uint16_t conn_latency;      /**< used slave latency for parent connection */
+    uint32_t conn_super_to;     /**< used supervision timeout for parent
+                                 *   connection */
+
+    /* time the node searches and ranks potential parents
+     * -> itvl := rand([eval_itvl:2*eval_itvl)) */
+    uint32_t eval_itvl_min;     /**< amount of time a node searches for
+                                 *   potential parents, lower bound */
+    uint32_t eval_itvl_max;     /**< amount of time a node searches for
+                                 *   potential parents, upper bound */
+} nimble_rpble_cfg_t;
+
+/**
+ * @brief   RPL DODAG information
+ */
+typedef struct {
+    uint8_t inst_id;            /**< instance ID */
+    uint8_t dodag_id[16];       /**< DODAG ID */
```

Comment:
not `ipv6_addr_t`?

Reply:
Unfortunately this will not work. When including `net/ipv6/addr.h` in `nimble_rpble.h` it will lead to collisions between RIOTs `byteorder.h` and NimBLE`s `byteorder.h` -> the nimble variant is pulled in by the `nimble_netif` dependency in the same file. So as of now, we need to be careful when missing NimBLE and RIOT deps...